app/src/main/java/com/andrew/tasky/agenda/data/event/EventRepositoryImpl.kt

```diff
@@ -1,33 +1,141 @@
+
+        if (isEventInDb || isModifiedCreateEvent) {
+            val result = getResourceResult {
+                api.createEvent(
+                    eventData = MultipartBody.Part
+                        .createFormData(
+                            name = "create_event_request",
+                            value = Json.encodeToString(compressedEvent.toCreateEventRequest())
+                        ),
+                    photoData = compressedEvent.photos.filterIsInstance<LocalPhoto>().mapIndexed {
+                        index, eventPhoto ->
+                        MultipartBody.Part
+                            .createFormData(
+                                name = "photo$index",
+                                filename = eventPhoto.key,
+                                body = eventPhoto.byteArray!!.toRequestBody()
+                            )
+                    }
+                )
+            }
+            return when (result) {
+                is Resource.Error -> {
+                    db.getEventDao().upsertModifiedEvent(
+                        ModifiedEventEntity(
+                            id = compressedEvent.id,
+                            modifiedType = ModifiedType.CREATE
+                        )
+                    )
+                    compressedEvent.photos.filterIsInstance<LocalPhoto>().forEach {
+                        saveLocalPhotoInternally(it)
+                        it.savedInternally = true
+                    }
+                    UpsertEventResult(photosTooBig = photosDeleted, resource = Resource.Error())
+                }
+                is Resource.Success -> {
+                    result.data?.let {
+                        db.getEventDao().upsertEvent(
+                            it.toEventEntity(
+                                isDone = compressedEvent.isDone,
+                                isGoing = compressedEvent.isGoing
+                            )
+                        )
+                    }
+                    UpsertEventResult(photosTooBig = photosDeleted, resource = Resource.Success())
+                }
+            }
+        } else {
+            val result = getResourceResult {
+                api.updateEvent(
+                    eventData = MultipartBody.Part
+                        .createFormData(
+                            name = "update_event_request",
+                            value = Json.encodeToString(compressedEvent.toUpdateEventRequest())
+                        ),
+                    photoData = compressedEvent.photos.filterIsInstance<LocalPhoto>().mapIndexed {
+                        index, eventPhoto ->
+                        MultipartBody.Part
+                            .createFormData(
+                                name = "photo$index",
+                                filename = eventPhoto.key,
+                                body = eventPhoto.byteArray!!.toRequestBody()
```

Comment:
Are you 100% sure `byteArray` will never be null here? A more careful approach would be this:
```
body = eventPhoto.byteArray?.toRequestBody() ?: return@mapIndexed
```
That way you just ignore it if it doesn't exist.